src/mono/wasm/runtime/crypto-worker.ts

```diff
@@ -0,0 +1,211 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { Module } from "./imports";
+import { assert } from "./types";
+
+let mono_wasm_crypto: {
+    channel: LibraryChannel
+    worker: Worker
+} | null = null;
+
+export function dotnet_browser_can_use_simple_digest_hash(): number {
+    return mono_wasm_crypto === null ? 0 : 1;
+}
+
+export function dotnet_browser_simple_digest_hash(ver: number, input_buffer: number, input_len: number, output_buffer: number, output_len: number): number {
+    assert(!!mono_wasm_crypto, "subtle crypto not initialized");
+
+    const msg = {
+        func: "digest",
+        type: ver,
+        data: Array.from(Module.HEAPU8.subarray(input_buffer, input_buffer + input_len))
+    };
+
+    const response = mono_wasm_crypto.channel.send_msg(JSON.stringify(msg));
+    const digest = JSON.parse(response);
+    if (digest.length > output_len) {
+        console.info("call_digest: about to throw!");
+        throw "DIGEST HASH: Digest length exceeds output length: " + digest.length + " > " + output_len;
+    }
+
+    Module.HEAPU8.set(digest, output_buffer);
+    return 1;
+}
+
+export function init_crypto(): void {
+    if (typeof globalThis.crypto !== "undefined" && typeof globalThis.crypto.subtle !== "undefined"
+        && typeof SharedArrayBuffer !== "undefined"
+        && typeof Worker !== "undefined"
+    ) {
+        console.debug("MONO_WASM: Initializing Crypto WebWorker");
+
+        const chan = LibraryChannel.create(1024); // 1024 is the buffer size in char units.
+        const worker = new Worker("dotnet-crypto-worker.js");
+        mono_wasm_crypto = {
+            channel: chan,
+            worker: worker,
+        };
+        worker.postMessage({
+            comm_buf: chan.get_comm_buffer(),
+            msg_buf: chan.get_msg_buffer(),
+            msg_char_len: chan.get_msg_len()
+        });
+        worker.onerror = event => {
+            console.warn(`MONO_WASM: Error in Crypto WebWorker. Cryptography digest calls will fallback to managed implementation. Error: ${event.message}`);
```

Comment:
I don't think we want this fallback. I thought it was all or nothing. @bartonjs?

Reply:
All or nothing was the original plan.  But since Safari can't (at least, couldn't) do it we got authorization to do native-preferred with managed fallback.  (That's why we already have managed SHA1/SHA256/SHA384/SHA512 checked in for wasm)

Reply:
Safari is always a no, right? The never okay I get and that check is at the top of this method. However, this error callback path seems to imply that if a catastrophic error occurs during a call where the SubtleCrypto libraries are supported, on the next call we will fallback to the managed implementation. The message might be confusing me, but setting `mono_wasm_crypto` to `null` after it was set appears we are falling back on a supported platform. Is that fallback also supported?

Reply:
Oooh.  Um, no, that probably shouldn't fall back.  I think that we're expecting to be able to say "these are the characteristics, and if your browser has them then you get native crypto (which may or may not be FIPS/etc-acceptable) over managed crypto (which never is)".  Changing after the first call seems... bad.

@blowdart or @GrabYourPitchforks may wish to disagree and say that if stuff breaks it's OK to fall back to managed... but I think that makes Barry's job harder, so I don't expect they'll feel differently.

Reply:
I seem to remember discussing this and talking about addressing it with documentation and no promises ever of fips compliance (I still hate it though). @GrabYourPitchforks does that ring a bell?

Reply:
Because of https://github.com/dotnet/runtime/pull/65966/files#diff-46676f417228802cba4f54049a643b76f11d4b8feb4953f39cb439ecaeba4097R10, I don't think this is a dynamic fallback... just the startup code.  But it'd be good to get full closure on this in the fullness of time.